### ISSUE TITLE
App crashes in In-App Messaging card type with empty message body

### ISSUE THREAD

**Issue by devnull-woowa** [firebase/firebase-android-sdk#1362]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 3.5.3
  * Firebase Component: In-App Messaging
  * Component version: 19.0.2

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

#### Relevant Code:

```
1. Compose a campaign with Card Type layout
2. Set optional message body to empty
3. App crashes
```

**Comment by devnull-woowa** [comment]
```
Caused by java.lang.IllegalArgumentException: Text model must have a color
       at com.google.firebase.inappmessaging.model.Text$Builder.build(Text.java:116)
       at com.google.firebase.inappmessaging.model.ProtoMarshallerClient.decode(ProtoMarshallerClient.java:202)
       at com.google.firebase.inappmessaging.model.ProtoMarshallerClient.from(ProtoMarshallerClient.java:116)
       at com.google.firebase.inappmessaging.model.ProtoMarshallerClient.a(ProtoMarshallerClient.java:224)
       at com.google.firebase.inappmessaging.internal.InAppMessageStreamManager.triggeredInAppMessage(InAppMessageStreamManager.java:304)
       at com.google.firebase.inappmessaging.internal.InAppMessageStreamManager.lambda$getTriggeredInAppMessageMaybe$27(InAppMessageStreamManager.java:298)
       at com.google.firebase.inappmessaging.internal.InAppMessageStreamManager$$Lambda$12.apply(InAppMessageStreamManager.java:6)
       at io.reactivex.internal.operators.maybe.MaybeFlatten$FlatMapMaybeObserver.onSuccess(MaybeFlatten.java:89)
       at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onNext(FlowableElementAtMaybe.java:80)
       at 
...

**Comment by cristianatolea** [comment]
I have the same problem. The app crashes with the above error log. I stopped all campaigns but that did not work, the app still crashes. I've also removed the in-app message library to check if that was the problem and the app worked fine without any reference to it.

**Comment by kameshyadavky** [comment]
I am having this problem too. The app crashes with the same log. Even stopping the campaign does not help. I have to clear the app's data. The crash seems to go away as soon as I fill the option body message in campaign.

**Comment by cristianatolea** [comment]
> I am having this problem too. The app crashes with the same log. Even stopping the campaign does not help. I have to clear the app's data. The crash seems to go away as soon as I fill the option body message in campaign. In-app-messaging version : 19.0.3

Are you sure you stopped all running campaigns? I thought the same, but then discovered there were some more campaigns running.

**Comment by kameshyadavky** [comment]
> > I am having this problem too. The app crashes with the same log. Even stopping the campaign does not help. I have to clear the app's data. The crash seems to go away as soon as I fill the option body message in campaign. In-app-messaging version : 19.0.3
> 
> Are you sure you stopped all running campaigns? I thought the same, but then discovered there were some more campaigns running.

Yes I stopped all the campaigns, have to clear the app data and filling the optional body field resolved the error.

**Comment by ccmtmp** [comment]
I am having this problem too.

**Comment by rosariopfernandes** [comment]
I was able to reproduce this on version 19.0.4 of the library. I'll send a PR with a fix

**Comment by devnull-woowa** [comment]
This is a serious issue. Once the campaign data is cached in the local cache, stopping or deleting the campaign is useless. It is not possible to induce forced updates because it tries repeatedly at the app start. We have no choice but to want users to clear their app data or update themselves to a version that resolves the issue or just wait until local caches to be expired.
I think It need a way to clear the local cache when the campaign is stopped

**Comment by mohit92** [comment]
Since this has started happening recently without any code change in the app, I am suspecting that something has changed on their backend. Their backend is not sending the data to clients in the expected format which is causing this crash.
I think there are two parts to this issue:
1. Fix the client SDK to handle the non empty string errors
2. Fix the backend at priority so that no new users are impacted.

**Comment by kanodianeha** [comment]
We are facing the same issue :(

**Comment by JasonAHeron** [comment]
Hey folks, we had a change to our backend that would have caused this issue bu we rolled it back. Let me reproduce and we will get this fixed today.

**Comment by JasonAHeron** [comment]
Quick followup here, I have reproduced the issue and I'm working on a backend fix now.

**Comment by JasonAHeron** [comment]
I have a fix rolling out to production now. I'll let you folks know this this is resolved.

**Comment by JasonAHeron** [comment]
Rollout for fix in progress, we estimate it will take about an hour for the rollout to complete

**Comment by JasonAHeron** [comment]
Hey folks, this is now fixed. For users who have a cached bad state it seems the best solution (as horrible as this is) is to uninstall and reinstall the app (or clear app data). Going forward I will be fixing the SDK to remove this class of errors in the future. Really sorry about this.

**Comment by rosariopfernandes** [comment]
@JasonAHeron Glad it is now fixed :) I had written a fix for this on the client side here: #1369 Please have a look and feel free to close it, if you find it not relevant anymore

**Comment by JasonAHeron** [comment]
Hey I appreciate the PR but I'm going to be applying a more general fix to all of our proto parsing to remove this entire class of errors rather than just this one error. I'm going to go ahead and close our your PR but I really appreciate you taking the time to come up with a fix.

I'm also closing out this issue as this is now resolved.
